tests/wsl/prepare_wsl_feature.pm

```diff
@@ -0,0 +1,95 @@
+    enable_wsl_feature => {
+        wsl         => q{Enable-WindowsOptionalFeature -Online -FeatureName "Microsoft-Windows-Subsystem-Linux" -NoRestart},
+        vm_platform => q{Enable-WindowsOptionalFeature -Online -FeatureName "VirtualMachinePlatform" -NoRestart}
+    }
+};
+
+sub run {
+    my $wsl_appx_filename = get_required_var('ASSET_1');
+    assert_screen 'windows-desktop';
+
+    # 0) Enable developer mode
+    open_powershell_as_admin;
+    run_in_powershell $powershell_cmds->{enable_developer_mode};
+
+    # 1) Enable WSL 1 or 2
+    run_in_powershell $powershell_cmds->{enable_wsl_feature}->{vm_platform} if (get_var('some_future_var') =~ m/wsl/);
+    run_in_powershell $powershell_cmds->{enable_wsl_feature}->{wsl};
+
+    # 2) Download the image
+    run_in_powershell 'Invoke-WebRequest -Uri ' . data_url('ASSET_1') . ' -O C:\\' . $wsl_appx_filename . ' -UseBasicParsing';
+
+    # 3) Open the image file in Explorer
+    run_in_powershell 'ii C:\\';
+    assert_and_click 'wsl-appx-file', timeout => 60, button => 'right';
+    wait_still_screen stilltime => 3, timeout => 10;
+
+    # 4) Reach certificate installation
+    assert_and_click 'open-file-properties';
+    assert_and_click 'digital-signatures';
+    assert_and_click 'build-service-cert';
+    wait_screen_change(sub { send_key 'alt-d' }, 10);
+    assert_and_click 'view-certificate-details';
+    assert_and_click 'install-certificate';
+    wait_still_screen stilltime => 3, timeout => 10;
+    assert_and_click 'install-certificate-to-local-machine';
+    wait_screen_change(sub { send_key 'ret' }, 10);
+    assert_screen 'user-acount-ctl-allow-make-changes', 20;
+    assert_and_click 'user-acount-ctl-yes';
+    wait_still_screen stilltime => 3, timeout => 10;
+    assert_and_click 'select-custom-store', timeout => 120;
+    wait_still_screen stilltime => 1, timeout => 5;
+    assert_and_click 'browse';
+    assert_and_click 'select-store-trusted-roots';
+    # confirm selected certificate store
+    wait_screen_change(sub { send_key 'ret' }, 10);
+    # hit next
+    wait_screen_change(sub { send_key 'ret' }, 10);
+    for (1 .. 2) {
+        assert_screen 'successful-certificate-import';
+        wait_screen_change(sub { send_key 'ret' }, 10);
+    }
+
+    # close all opened windows, including powershell
+    send_key_until_needlematch 'powershell-ready-prompt', 'alt-f4', 25, 2;
+    # close
+    type_string "exit", max_interval => 125;
+    send_key 'ret';
+    assert_screen 'windows-desktop';
+
+    # 5) reboot the SUT
+    reboot_or_shutdown('reboot');
+    wait_boot_windows;
+
+    # 6) Install Linux in WSL
+    open_powershell_as_admin;
+    type_string "C:\\$wsl_appx_filename";
+    wait_still_screen stilltime => 3, timeout => 10;
+    send_key 'ret';
+    wait_still_screen stilltime => 3, timeout => 10;
+}
+
+sub post_fail_hook { sleep 120; save_screenshot; }
```

Comment:
If you extract the common functions out of windows_utils and move them to a `windowsbasetest` class and use that as base, and add the post_fail_hook there, you could remove this one (and others).

Reply:
Hi Santy, thanks for the hint. I was thinking to convert `windows_utils` to `windowsbasetest`, I will do it now